src/D2L.Bmx/AwsCredsCreator.cs

```diff
@@ -1,22 +1,30 @@
 using D2L.Bmx.Aws;
-using D2L.Bmx.Okta;
 using D2L.Bmx.Okta.Models;
 
 namespace D2L.Bmx;
 
 internal class AwsCredsCreator(
 	IAwsClient awsClient,
 	IConsolePrompter consolePrompter,
+	IAwsCredentialCache awsCredentialCache,
 	BmxConfig config
 ) {
 	public async Task<AwsCredentials> CreateAwsCredsAsync(
-		IOktaApi oktaApi,
+		AuthenticatedOktaApi oktaApi,
 		string? account,
 		string? role,
 		int? duration,
-		bool nonInteractive
+		bool nonInteractive,
+		bool cache
 	) {
-		OktaApp[] awsApps = await oktaApi.GetAwsAccountAppsAsync();
+		if( cache && !File.Exists( BmxPaths.CONFIG_FILE_NAME ) ) {
+			throw new BmxException(
+				"BMX global config file not found. Will not cache credentials on shared use machines."
+			);
+		}
+
+		//cache this one?
+		OktaApp[] awsApps = await oktaApi.Api.GetAwsAccountAppsAsync();
```

Comment:
Good question, I would think we should be doing this, out of scope of this PR.

Realistically I feel like we are more at risk of hitting Okta limits (not necessarily rate limits but rather any DoS/Brute forcing prevention heuristics) than AWS STS limits.

In fact that's the only limit I recall someone (Mark Tse) hitting while using BMX in a script

Reply:
Pretty sure we don't want to hit Okta when we have cached creds.
Don't think we need to cache this though. Can we look up the AWS creds cache first and return early?

Reply:
Hmm, good point. I think so? We can move the return early w/ cache hit here and proceed to the querying the Okta API + prompt user to select a role (if they aren't using ``--non-interactive``)  & there is no match for the given ``--role`` etc..